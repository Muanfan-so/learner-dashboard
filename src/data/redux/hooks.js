import { useSelector } from 'react-redux';

import { actions as appActions } from './app/reducer';
import appSelectors from './app/selectors';
import requestSelectors from './requests/selectors';

const { courseCard } = appSelectors;

export const usePageNumber = () => useSelector(appSelectors.pageNumber);
export const useEmailConfirmationData = () => useSelector(appSelectors.emailConfirmation);
export const useEnterpriseDashboardData = () => useSelector(appSelectors.enterpriseDashboard);
export const usePlatformSettingsData = () => useSelector(appSelectors.platformSettings);
// suggested courses is max at 3 at the moment.
export const useSuggestedCoursesData = () => useSelector(appSelectors.suggestedCourses).slice(0, 3);
export const useRecommendedCoursesData = () => useSelector(appSelectors.recommendedCourses);
export const useSelectSessionModalData = () => useSelector(appSelectors.selectSessionModal);

export const useHasCourses = () => useSelector(appSelectors.hasCourses);
export const useHasAvailableDashboards = () => useSelector(appSelectors.hasAvailableDashboards);
export const useCurrentCourseList = (opts) => useSelector(
  state => appSelectors.currentList(state, opts),
);
export const useShowSelectSessionModal = () => useSelector(appSelectors.showSelectSessionModal);

// eslint-disable-next-line
export const useCourseCardData = (selector) => (cardId) => useSelector(
  (state) => selector(state, cardId),
);

export const useCardCertificateData = useCourseCardData(courseCard.certificate);
export const useCardCourseData = useCourseCardData(courseCard.course);
export const useCardCourseRunData = useCourseCardData(courseCard.courseRun);
export const useCardEnrollmentData = useCourseCardData(courseCard.enrollment);
export const useCardEntitlementData = useCourseCardData(courseCard.entitlement);
export const useCardGradeData = useCourseCardData(courseCard.gradeData);
export const useCardProviderData = useCourseCardData(courseCard.courseProvider);
export const useCardRelatedProgramsData = useCourseCardData(courseCard.relatedPrograms);

export const useUpdateSelectSessionModalCallback = (dispatch, cardId) => () => dispatch(
  appActions.updateSelectSessionModal(cardId),
);

export const useMasqueradeData = () => useSelector(requestSelectors.masquerade);

export const useIsPendingRequest = (requestName) => useSelector(requestSelectors.isPending(requestName));
