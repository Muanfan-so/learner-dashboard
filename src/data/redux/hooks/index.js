import * as app from './app';
import * as requests from './requests';

export * from './app';
export { requests };

export default { app, requests };
