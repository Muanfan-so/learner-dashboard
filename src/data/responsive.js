import { useWindowSize, breakpoints } from '@edx/paragon';
import { StrictDict } from 'utils';

const createResponsive = (fn) => () => {
  const { width } = useWindowSize();
  return fn(width);
};

export const isDesktopSize = createResponsive(
  (width) => width >= breakpoints.large.minWidth,
);

// Beware of using isPadSize() for styling. When the screen resizes pass the
// breakpoint, the screen will flash between the two styles. This is why it is
// recommened isDesktopSize() or isMobileSize() be used instead.
export const isIpadSize = createResponsive(
  (width) => width >= breakpoints.medium.minWidth && width < breakpoints.large.minWidth,
);

export const isMobileSize = createResponsive(
  (width) => width < breakpoints.medium.minWidth,
);

export default StrictDict({
  isDesktopSize,
  isIpadSize,
  isMobileSize,
});
