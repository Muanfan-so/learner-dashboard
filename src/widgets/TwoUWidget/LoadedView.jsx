import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { useIntl } from '@edx/frontend-platform/i18n';
import { Badge, Hyperlink, Icon } from '@edx/paragon';
import { ArrowForward } from '@edx/paragon/icons';

import { useIsMediumScreen } from './hooks';
import { getMerchandisingItems, countryCodeUS } from './constants';
import messages from './messages';

export const LoadedView = ({ countryCode, show2ULobs }) => {
  const { formatMessage } = useIntl();
  const merchandisingItems = getMerchandisingItems(countryCode);
  const isMediumScreen = useIsMediumScreen();
  const markup = { markup: children => <span className="d-inline-flex mb-0 text-brand-500">{children}</span> };
  const items = merchandisingItems.map(item => (
    <Hyperlink
      destination={item.itemURL}
      key={item.key}
      target="_blank"
      showLaunchIcon={false}
      className={classNames('static-callouts-item', {
        'static-callouts-item-md': isMediumScreen && countryCode === countryCodeUS,
      })}
    >
      <Badge variant="warning" className="item-badge mb-2">{formatMessage(messages.badgeTitle)}</Badge>
      <h3>
        {formatMessage(item.title)}
      </h3>
      <div className="card-description">
        {formatMessage(item.description)}
        <Icon className="d-inline-block align-middle mx-1" src={ArrowForward} />
      </div>
    </Hyperlink>
  ));

  return (
    show2ULobs && (
      <div className="callouts-container">
        <div>
          <h2>
            {formatMessage(messages.heading, markup)}
          </h2>
          <p className="mb-0 text-gray-700">
            {formatMessage(messages.subheading)}
          </p>
        </div>
        <div className={classNames('static-callouts-container', {
          'static-callouts-container-md': isMediumScreen && countryCode === countryCodeUS,
        })}
        >
          {items}
        </div>
      </div>
    )
  );
};

LoadedView.propTypes = {
  countryCode: PropTypes.string.isRequired,
  show2ULobs: PropTypes.bool.isRequired,
};

export default LoadedView;
