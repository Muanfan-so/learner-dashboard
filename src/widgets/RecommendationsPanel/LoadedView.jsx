import React from 'react';
import PropTypes from 'prop-types';

import { useIntl } from '@edx/frontend-platform/i18n';
import { Button } from '@edx/paragon';
import { Search } from '@edx/paragon/icons';

import { hooks } from 'data/redux';
import CourseCard from './components/CourseCard';
import messages from './messages';

import './index.scss';

export const LoadedView = ({
  courses,
  isPersonalizedRecommendation,
  courseSearchClickTracker,
}) => {
  const { courseSearchUrl } = hooks.usePlatformSettingsData();
  const { formatMessage } = useIntl();

  return (
    <div className="p-4 w-100 panel-background">
      <h3 className="pb-2">{isPersonalizedRecommendation
        ? formatMessage(messages.recommendationsHeading) : formatMessage(messages.popularCoursesHeading)}
      </h3>
      <div>
        {courses.map((course) => (
          <CourseCard
            key={course.courseKey}
            course={course}
            isPersonalizedRecommendation={isPersonalizedRecommendation}
          />
        ))}
      </div>
      <div className="text-center explore-courses-btn">
        <Button
          variant="tertiary"
          iconBefore={Search}
          as="a"
          href={courseSearchUrl}
          onClick={courseSearchClickTracker}
        >
          {formatMessage(messages.exploreCoursesButton)}
        </Button>
      </div>
    </div>
  );
};

LoadedView.propTypes = {
  courses: PropTypes.arrayOf(PropTypes.shape({
    courseKey: PropTypes.string,
    title: PropTypes.string,
    logoImageUrl: PropTypes.string,
    marketingUrl: PropTypes.string,
  })).isRequired,
  isPersonalizedRecommendation: PropTypes.bool.isRequired,
  courseSearchClickTracker: PropTypes.func.isRequired,
};

export default LoadedView;
