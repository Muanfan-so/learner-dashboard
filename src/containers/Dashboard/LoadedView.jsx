import React from 'react';
import classNames from 'classnames';
import {
  Container,
  Col,
  Row,
} from '@edx/paragon';

import { hooks as appHooks } from 'data/redux';
import { RequestKeys } from 'data/constants/requests';
import CourseList from 'containers/CourseList';
import RecommendationsPanel, { LoadingView as RecommendationsLoadingView } from 'containers/RecommendationsPanel';
import WidgetSidebar from 'containers/WidgetSidebar';
import hooks from './hooks';

const commonColumnConfig = {
  sm: { span: 12, offset: 0 },
  md: { span: 10, offset: 1 },
  lg: { span: 12, offset: 0 },
};

export const columnConfig = {
  courseList: {
    ...commonColumnConfig,
    xl: { span: 8, offset: 0 },
  },
  sidebar: {
    ...commonColumnConfig,
    xl: { span: 4, offset: 0 },
  },
};

export const LoadedView = () => {
  const isCollapsed = hooks.useIsDashboardCollapsed();
  const recommendedCourses = appHooks.useRecommendedCoursesData();
  const hasRecommendedCourses = recommendedCourses.courses.length > 0;
  const isRecommendationsPending = appHooks.useRequestIsPending(RequestKeys.recommendedCourses);

  return (
    <Container fluid size="xl">
      <Row>
        <Col {...columnConfig.courseList} className="course-list-column">
          <CourseList />
        </Col>
        <Col {...columnConfig.sidebar} className="sidebar-column">
          {!isCollapsed && (<h2 className="course-list-title">&nbsp;</h2>)}
          <WidgetSidebar />
        </Col>
      </Row>
    </Container>
  );
};

export default LoadedView;
