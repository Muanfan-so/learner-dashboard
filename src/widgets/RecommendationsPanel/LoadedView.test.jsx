import React from 'react';
import { shallow } from 'enzyme';

import { hooks } from 'data/redux';
import LoadedView from './LoadedView';
import mockData from './mockData';

jest.mock('./components/CourseCard', () => 'CourseCard');
jest.mock('data/redux', () => ({
  hooks: {
    usePlatformSettingsData: jest.fn(),
  },
}));

const courseSearchUrl = 'test-course-search-url';
hooks.usePlatformSettingsData.mockReturnValue(courseSearchUrl);

describe('RecommendationsPanel LoadedView', () => {
  test('snapshot', () => {
    expect(shallow(<LoadedView courses={mockData.courses} />)).toMatchSnapshot();
  });
});
