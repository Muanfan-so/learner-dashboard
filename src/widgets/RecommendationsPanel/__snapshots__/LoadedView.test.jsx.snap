// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecommendationsPanel LoadedView snapshot 1`] = `
<div
  className="p-4 panel-background"
>
  <h3
    className="pb-2"
  >
    Recommendations for you
  </h3>
  <div>
    <CourseCard
      course={
        Object {
          "courseKey": "cs-1",
          "logoImageUrl": "https://prod-discovery.edx-cdn.org/media/programs/banner_images/9a310b98-8f27-439e-be85-12d6460245c9-f2efca129273.small.jpg",
          "marketingUrl": "www.edx/recommended-course",
          "title": "Recommended course 1",
        }
      }
      key="cs-1"
    />
    <CourseCard
      course={
        Object {
          "courseKey": "cs-2",
          "logoImageUrl": "https://prod-discovery.edx-cdn.org/media/programs/banner_images/9a310b98-8f27-439e-be85-12d6460245c9-f2efca129273.small.jpg",
          "marketingUrl": "www.edx/recommended-course",
          "title": "Recommended course 2 with a really really really long name for some reason",
        }
      }
      key="cs-2"
    />
    <CourseCard
      course={
        Object {
          "courseKey": "cs-3",
          "logoImageUrl": "https://prod-discovery.edx-cdn.org/media/programs/banner_images/9a310b98-8f27-439e-be85-12d6460245c9-f2efca129273.small.jpg",
          "marketingUrl": "www.edx/recommended-course",
          "title": "Recommended course 3",
        }
      }
      key="cs-3"
    />
    <CourseCard
      course={
        Object {
          "courseKey": "cs-4",
          "logoImageUrl": "https://prod-discovery.edx-cdn.org/media/programs/banner_images/9a310b98-8f27-439e-be85-12d6460245c9-f2efca129273.small.jpg",
          "marketingUrl": "www.edx/recommended-course",
          "title": "Recommended course 4",
        }
      }
      key="cs-4"
    />
  </div>
  <div
    className="text-center explore-courses-btn"
  >
    <Button
      as="a"
      variant="tertiary"
    >
      Explore courses
    </Button>
  </div>
</div>
`;
